### PR TITLE
Ctw 474/in progress info

### PR DESCRIPTION
**Pull request by pmadria3**
CTW-474

This PR adds in-progress/initialize loading to the form drawer. 
Screenshot 2023-01-30 at 4 52 50 PM

[CTW-474]: https://zeushealth.atlassian.net/browse/CTW-474?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/content/patient-history/patient-history-message-status.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            className="ctw-h-5 ctw-flex-none ctw-fill-caution-icon"
+          >
+            <path
+              d="M304 48c0-26.5-21.5-48-48-48s-48 21.5-48 48s21.5 48 48 48s48-21.5 48-48zm0 
+            416c0-26.5-21.5-48-48-48s-48 21.5-48 48s21.5 48 48 48s48-21.5 48-48zM48 304c26.5 0 48-21.5 48-48s-21.5-48-48-48s-48
+             21.5-48 48s21.5 48 48 48zm464-48c0-26.5-21.5-48-48-48s-48 21.5-48 48s21.5 48 48 48s48-21.5 48-48zM142.9 437c18.7-18.7
+              18.7-49.1 0-67.9s-49.1-18.7-67.9 0s-18.7 49.1 0 67.9s49.1 18.7 67.9 0zm0-294.2c18.7-18.7 18.7-49.1 0-67.9S93.7 56.2 
+              75 75s-18.7 49.1 0 67.9s49.1 18.7 67.9 0zM369.1 437c18.7 18.7 49.1 18.7 67.9 0s18.7-49.1 0-67.9s-49.1-18.7-67.9 0s-18.7 49.1 0 67.9z"
+            />
+          </svg>
```
Comment (luthfib) [inline src/components/content/patient-history/patient-history-message-status.tsx:29]: Can this go into its own icon file?

File: src/fhir/formatters.ts [hunk 2]
```diff
@@ -108,3 +108,14 @@ export function formatAge(age: fhir4.Age): string {
   const { value, unit } = age;
   return compact([value, unit]).join(" ");
 }
+
+// Formats a string ISO date to MM/DD/YYY.
+export function formatStringISOtoDate(dateStr?: string): string | undefined {
```
Comment (luthfib) [inline src/fhir/formatters.ts:113]: Dont we already like this with `formatStringToDate`?

Comment (pmadria3) [reply src/fhir/formatters.ts:113]: So we do but this format is different - that one formats a string yyyymmdd or yyyyMMddHHmmss into MM/DD/YYYY. This one is in ISO form with dashes.

File: src/services/patient-history/patient-history.ts [hunk 3]
```diff
@@ -33,18 +33,55 @@ export async function getPatientRefreshHistoryMessages(
...
+        patientHistoryExists: true,
+        status: "error",
+        dateCreated: messages[0]._createdAt,
+      };
+    default:
+      return {
+        patientHistoryExists: false,
+        status: "",
+        dateCreated: messages[0]._createdAt,
+      };
```
Comment (luthfib) [inline src/services/patient-history/patient-history.ts:85]: This switch seems redundant. you dont really need `patientHistoryExists` as you are always returning true as long as it has a status and you are always setting status to `messages[0].status`

**Review by luthfib — COMMENTED** [review]
Looking, good but left some comments

File: src/services/patient-history/patient-history.ts [hunk 4]
```diff
@@ -33,18 +33,56 @@ export async function getPatientRefreshHistoryMessages(
...
+        ...lastPatientHistoryRequest,
+      };
+    case "initialize":
+      return {
+        patientHistoryExists: false,
+        ...lastPatientHistoryRequest,
+      };
+    case "error":
+      return {
+        patientHistoryExists: true,
```
Comment (WesleyKapow) [inline src/services/patient-history/patient-history.ts:78]: true for error?

File: src/services/patient-history/patient-history.ts [hunk 5]
```diff
@@ -33,18 +33,56 @@ export async function getPatientRefreshHistoryMessages(
...
+    case "in_progress":
+      return {
+        patientHistoryExists: false,
+        ...lastPatientHistoryRequest,
+      };
+    case "initialize":
+      return {
+        patientHistoryExists: false,
+        ...lastPatientHistoryRequest,
+      };
```
Comment (WesleyKapow) [inline src/services/patient-history/patient-history.ts:75]: These are the same ya? You can take advantage of switch fallthrough:

```
case "in_progress":
case "initialize":
  return ...
```

**Review by luthfib — DISMISSED** [review]
LGTM

File: src/services/patient-history/patient-history.ts [hunk 6]
```diff
@@ -33,18 +33,56 @@ export async function getPatientRefreshHistoryMessages(
...
+
+  switch (messages[0].status) {
+    case "done":
+      return {
+        patientHistoryExists: true,
+        ...lastPatientHistoryRequest,
+      };
+    case "in_progress":
+      return {
+        patientHistoryExists: false,
```
Comment (WesleyKapow) [inline src/services/patient-history/patient-history.ts:68]: How is `patientHistoryExists` used? Couldn't there be another message that has it as `done`?

File: .changeset/soft-trees-kick.md [hunk 7]
```diff
@@ -0,0 +1,5 @@
+---
+"@zus-health/ctw-component-library": minor
+---
+
+Add progress bar in patient history drawer.
```
Comment (luthfib) [inline .changeset/soft-trees-kick.md:5]:
```suggestion
Add  in-progress state in patient history drawer.
```

File: src/fhir/formatters.ts (hunk 2 above)
Comment (WesleyKapow) [inline src/fhir/formatters.ts:113]: nit: name suggestion to follow existing pattern: `formatISODateStringToDate`

File: .changeset/soft-trees-kick.md [hunk 8]
```diff
@@ -0,0 +1,5 @@
+---
+"@zus-health/ctw-component-library": minor
+---
+
+Add  in-progress state in patient history drawer.
```
Comment (luthfib) [inline .changeset/soft-trees-kick.md:5]:
```suggestion
Add in-progress banner in patient history drawer.
```

File: src/components/content/patient-history/patient-history-message-status.tsx [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+            <div className="ctw-text-caution-message">
+              In Progress - request received {formatStringISOtoDate(date)}
+            </div>
+          </div>
+        </div>
+      );
+    case "done":
+    case "error":
+    default:
+      return <div />;
```
Comment (WesleyKapow) [inline src/components/content/patient-history/patient-history-message-status.tsx:29]: Do we actually want an empty div or could this return `null` instead?

File: src/components/content/patient-history/patient-history-message-status.tsx [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+  status,
+  date,
+}: PatientHistoryStatusProps) => {
+  switch (status) {
+    case "initialize":
+    case "in_progress":
+      return (
+        <div className="ctw-alert-bg ctw-rounded-md ctw-bg-caution-bg">
+          <LoadingIcon className="ctw-h-5 ctw-flex-none ctw-fill-caution-icon" />
+          <div className="ctw-space-y-2">
```
Comment (WesleyKapow) [inline src/components/content/patient-history/patient-history-message-status.tsx:19]: Spacing class only works when there are 2 or more children. I think you can remove this div.

**Review by luthfib — DISMISSED** [review]

File: src/components/content/conditions.tsx [hunk 11]
```diff
@@ -351,10 +362,16 @@ export function Conditions({ className, readOnly = false }: ConditionsProps) {
       {patientResponse.data && (
         <PatientHistoryRequestDrawer
           header={
-            <div className="ctw-pt-0 ctw-text-base">
-              Request patient clinical history from 70K+ providers across the
-              nation. No changes will be made to your patient record.
-            </div>
+            <>
```
Comment (WesleyKapow) [inline src/components/content/conditions.tsx:365]: Should there be a set space between the two child divs? If so, you would set this to a div and add `space-y-*`.

**Review by WesleyKapow — DISMISSED** [review]

File: src/services/patient-history/patient-history.ts (hunk 4 above)
Comment (pmadria3) [reply src/services/patient-history/patient-history.ts:78]: Yeah so there's a weird thing where there is a master status and then statuses of the two connectors (Surescripts and Commonwell) and so if one of them is an error but the other passes -> the master status is error. I met with michelle to see if we wanted to create some sort of map to make see what happens and she said to just focus on the master status and work off of that. So you can have a state where there is an error message that shows up but the data still shows in the table as one one connector passes and the other fails (as in the case with this test patient).

File: src/services/patient-history/patient-history.ts (hunk 5 above)
Comment (pmadria3) [reply src/services/patient-history/patient-history.ts:75]: They are the same but the `status` will be different for each of them depending on the case

Comment (pmadria3) [reply src/services/patient-history/patient-history.ts:75]: one of the fields in the `lastPatientHistoryRequest`

File: src/services/patient-history/patient-history.ts (hunk 6 above)
Comment (pmadria3) [reply src/services/patient-history/patient-history.ts:68]: Not sure what this means but I use patientHistoryExists in the conditions to deal with null states and if it exists and when to show in `conditions.tsx`. status is what contains the `done` and is passed down. (In the future there will be more banners so we will need the date and status)

File: src/components/core/loading-icon.tsx [hunk 12]
```diff
@@ -0,0 +1,21 @@
+export type LoadingIconProps = {
```
[the remark this answers is not in the document]
Comment (pmadria3) [reply src/components/core/loading-icon.tsx:1]: Confirmed from UX to have 2 separate ones and renaming to `ProgressIcon`

**Pull request merged**